Packs/ServiceNow/Integrations/ServiceNowEventCollector/ServiceNowEventCollector.py

```diff
@@ -50,29 +59,124 @@ def get_audit_logs(self, from_time: str, limit: Optional[int] = None, offset: in
+    if not start_timestamp:
+        current_time = datetime.utcnow()
+        first_fetch_time = current_time - timedelta(minutes=1)
+        first_fetch_str = first_fetch_time.strftime(LOGS_DATE_FORMAT)
+        from_date = first_fetch_str
+    else:
+        from_date = start_timestamp
+
+    return from_date
+
+
+def add_time_field(events: List[Dict[str, Any]], log_type) -> List[Dict[str, Any]]:
+    """
+    Add a '_time' field to each event and set the source log type.
+
+    Args:
+        events (List[Dict[str, Any]]): List of events to add the '_time' field to.
+        log_type (str): Type of log to set as the 'source_log_type' for each event.
+
+    Returns:
+        List[Dict[str, Any]]: The list of events with '_time' and 'source_log_type' fields added.
     """
     for event in events:
         event["_time"] = datetime.strptime(event["sys_created_on"], LOGS_DATE_FORMAT).strftime(DATE_FORMAT)
+        event["source_log_type"] = log_type
+
     return events
 
 
-def process_and_filter_events(events: list, previous_run_ids: set, from_date: str):
+def get_limit(args: dict, client: Client):
+    """
+    Retrieve the limit for the number of logs to fetch, with defaults based on client settings.
+
+    Args:
+        args (dict): Dictionary of arguments potentially containing a "limit" key.
+        client (Client): Client instance with attributes for default fetch limits.
+
+    Returns:
+        int: The limit for the number of logs to fetch.
+    """
+    limit = arg_to_number(args.get("limit")) or client.fetch_limit_audit or 1000
+    return limit
+
+
+def process_and_filter_events(events: list, previous_run_ids: set, from_date: str, log_type: str):
     """
-    Removing duplicates and creating a set of last fetched ids with the same time.
+    Remove duplicates from events and create a set of last fetched IDs with the same timestamp.
+
+    Args:
+        events (list): List of events fetched from the API.
+        previous_run_ids (set): Set of event IDs matching the timestamp in the 'from_date' parameter.
+        from_date (str): Starting date for fetching events, based on the last run's timestamp.
+        log_type (str): Type of log to set as the 'source_log_type' for each event.
 
-    :param events: events fetched from the API
-    :param previous_run_ids: ids with time as the one in the from date param
-    :param from_date: from date from last_run object
-    :return: all unique events and a set of last ids of events with same time.
+    Returns:
+        tuple: A list of unique events and a set of the last fetched event IDs with the same timestamp.
     """
     unique_events = []
     from_date_datetime = datetime.strptime(from_date, LOGS_DATE_FORMAT)
     for event in events:
         create_time = datetime.strptime(event.get("sys_created_on"), LOGS_DATE_FORMAT)
+        event["_time"] = create_time.strftime(DATE_FORMAT)
+        event["source_log_type"] = log_type
         if event.get("sys_id") in previous_run_ids:
```

Comment:
Please save a list of duplicates and if there is something in this list in the end of the function demisto.log the duplicate ids that are filtered out.